es/10_1_Understanding_the_Foundation_of_P2SH.md

```diff
@@ -0,0 +1,99 @@
+        "hex": "a914a5d106eb8ee51b23cf60d8bd98bc285695f233f387",
+        "reqSigs": 1,
+        "type": "scripthash",
+        "addresses": [
+          "2N8MytPW2ih27LctLjn6LfLFZZb1PFSsqBr"
+        ]
+      }
+```
+El script de bloqueo es bastante simple: `OP_HASH160 a5d106eb8ee51b23cf60d8bd98bc285695f233f3 OP_EQUAL`. Como de costumbre, hay una gran cantidad de datos en el medio. Este es un hash de otro script de bloqueo oculto (`redeemScript`) que solo se revelará cuando se canjeen los fondos. En otras palabras, el script de bloqueo estándar para una dirección P2SH es: `OP_HASH160 <redeemScriptHash> OP_EQUAL`.
+
+> :book: ***¿Qué es un redeemScript?*** Cada transacción P2SH lleva la huella digital de un script de bloqueo oculto dentro de ella como un hash de 20 bytes. Cuando se canjea una transacción P2SH, el `redeemScript` completo (sin hash) se incluye como parte del `scriptSig`. Bitcoin se asegurará de que el `redeemScript` coincida con el hash; luego ejecuta el `redeemScript` para ver si los fondos se pueden gastar (o no).
+
+ ¡Uno de los elementos interesantes de las transacciones P2SH es que ni el remitente ni Blockchain saben realmente qué es el `redeemScript`! Un remitente simplemente envía a una dirección P2SH estandarizada marcada con un prefijo "2" y no se preocupa por cómo el destinatario va a recuperar los fondos al final.
+
+> :link: **TESTNET vs MAINNET:** en testnet, el prefijo para las direcciones P2SH es `2`, mientras que en mainnet, es `3`.
+
+## Comprender Cómo Crear un Script P2SH
+
+Dado que el script de bloqueo visible para una transacción P2SH es tan simple, crear una transacción de este tipo también es bastante simple. En teoria. Todo lo que necesita hacer es crear una transacción cuyo script de bloqueo incluya un hash de 20 bytes del `redeemScript`. Ese hash se realiza con el estándar de Bitcoin `OP_HASH160`.
+
+> :book: ***Qué es OP_HASH160?*** La operación hash estándar para Bitcoin realiza un hash SHA-256, luego un HASH RIPEMD-160.
+
+En general, se requieren cuatro pasos:
+
+1. Cree un script de bloqueo arbitrario con Bitcoin Script.
+2. Cree una versión serializada de ese script de bloqueo.
+3. Realice un hash SHA-256 en esos bytes serializados.
+4. Realice un hash RIPEMD-160 en los resultados de ese hash SHA-256.
+
+Cada uno de esos pasos, por supuesto, requiere algo de trabajo por sí solo, y algunos de ellos pueden ser bastante intrincados. La buena noticia es que realmente no tiene que preocuparse por por ellos, porque son lo sucifientemente complejos como para que normalmente tenga una API que se encargue de todo por usted. 
+
+Entonces, por ahora, solo le proporcionaremos una descripción general, para que comprenda la metodología general. Enn [§10.2: Construyendo la Estructura de P2SH](10_2_Building_the_Structure_of_P2SH.md) proporcionaremos una mirada más profunda a la creación de scripts, en caso de que alguna vez quiera comprender las entrañas de este proceso.
```

Comment:
Entonces, por ahora, solo le proporcionaremos una descripción general, para que comprenda la metodología general. En [

Reply:
I can´t see the typo on this one